src/contractTest/java/uk/gov/hmcts/reform/wataskmanagementapi/provider/service/TaskManagementGetTaskByIdPactTest.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.provider.service;
+
+import au.com.dius.pact.provider.junit5.PactVerificationContext;
+import au.com.dius.pact.provider.junit5.PactVerificationInvocationContextProvider;
+import au.com.dius.pact.provider.junitsupport.IgnoreNoPactsToVerify;
+import au.com.dius.pact.provider.junitsupport.Provider;
+import au.com.dius.pact.provider.junitsupport.State;
+import au.com.dius.pact.provider.junitsupport.loader.PactBroker;
+import au.com.dius.pact.provider.junitsupport.loader.VersionSelector;
+import au.com.dius.pact.provider.spring.junit5.MockMvcTestTarget;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.wataskmanagementapi.auth.access.AccessControlService;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.TaskController;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.task.Task;
+import uk.gov.hmcts.reform.wataskmanagementapi.services.CamundaService;
+
+import java.time.ZonedDateTime;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(SpringExtension.class)
+@Provider("wa_task_management_api_get_task_by_id")
+@PactBroker(scheme = "${PACT_BROKER_SCHEME:http}",
+    host = "${PACT_BROKER_URL:localhost}", port = "${PACT_BROKER_PORT:9292}", consumerVersionSelectors = {
+    @VersionSelector(tag = "master")})
+@Import(TaskManagementProviderTestConfiguration.class)
+@IgnoreNoPactsToVerify
+public class TaskManagementGetTaskByIdPactTest {
+
+    @Autowired
+    private AccessControlService accessControlService;
+
+    @Autowired
+    private CamundaService camundaService;
+
+    @TestTemplate
+    @ExtendWith(PactVerificationInvocationContextProvider.class)
+    void pactVerificationTestTemplate(PactVerificationContext context) {
+        if (context != null) {
+            context.verifyInteraction();
+        }
+    }
+
+    @BeforeEach
+    void beforeCreate(PactVerificationContext context) {
+        MockMvcTestTarget testTarget = new MockMvcTestTarget();
+        testTarget.setControllers(new TaskController(
+            camundaService,
+            accessControlService
+        ));
+        if (context != null) {
+            context.setTarget(testTarget);
+        }
+
+    }
+
+    @State({"appropriate task is returned"})
+    public void getTaskById() {
+        setInitiMock();
+    }
```

Comment:
I'm not enterily sure how yet but we should also make sure they are sending in valid authentication headers e.g
Given a caseworker user exists 
And I call from a whitelisted service 
Then the targetted task is returned. 

This will help when writing some contract tests for 403 and 401 errors if we need to cover those